Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -99,9 +103,69 @@ void Processor::assume_context(Thread& thread, FlatPtr flags)
 
 FlatPtr Processor::init_context(Thread& thread, bool leave_crit)
 {
-    (void)thread;
-    (void)leave_crit;
-    TODO_AARCH64();
+    VERIFY(g_scheduler_lock.is_locked());
+    if (leave_crit) {
+        // Leave the critical section we set up in Process::exec,
+        // but because we still have the scheduler lock we should end up with 1
+        VERIFY(in_critical() == 2);
+        m_in_critical = 1; // leave it without triggering anything or restoring flags
+    }
+
+    u64 kernel_stack_top = thread.kernel_stack_top();
+
+    // Add a random offset between 0-256 (16-byte aligned)
+    kernel_stack_top -= round_up_to_power_of_two(get_fast_random<u8>(), 16);
```

Comment:
This looks like some code in here could be shared with x86_64 too…

(…but it's not actually all _that_ much. So probably fine as is.)

Reply:
Yep, this should definitely be shared too. We just don't have that architecture at the moment to be able to easily share this code. It's on my :yaklist: :)